controller/program/tests/tests/rebalance/test_utils.rs

```diff
@@ -0,0 +1,502 @@
+        },
+        start::{
+            NewStartRebalanceIxPreAccsBuilder, StartRebalanceIxData, StartRebalanceIxPreKeysOwned,
+        },
+    },
+    keys::{INSTRUCTIONS_SYSVAR_ID, LST_STATE_LIST_ID, POOL_STATE_ID, REBALANCE_RECORD_ID},
+    ID,
+};
+use inf1_svc_ag_core::{
+    inf1_svc_lido_core::solido_legacy_core::TOKENKEG_PROGRAM,
+    inf1_svc_wsol_core::instructions::sol_val_calc::WsolCalcAccs, instructions::SvcCalcAccsAg,
+    SvcAgTy,
+};
+use inf1_svc_jiminy::traits::SolValCalcAccs;
+use inf1_test_utils::{
+    acc_bef_aft, assert_diffs_lst_state_list, assert_diffs_pool_state, create_pool_reserves_ata,
+    create_protocol_fee_accumulator_ata, keys_signer_writable_to_metas, lst_state_list_account,
+    mock_mint, mock_token_acc, pool_state_account, raw_mint, raw_token_acc, upsert_account, Diff,
+    DiffLstStateArgs, DiffsPoolStateArgs, LstStateData, LstStateListChanges, LstStateListData,
+    NewPoolStateBoolsBuilder, PkAccountTup, ALL_FIXTURES, JUPSOL_FIXTURE_LST_IDX, WSOL_MINT,
+};
+use jiminy_sysvar_instructions::sysvar::OWNER_ID;
+use jiminy_sysvar_rent::Rent;
+use sanctum_system_jiminy::sanctum_system_core::ID as SYSTEM_PROGRAM_ID;
+use solana_account::Account;
+use solana_instruction::{AccountMeta, BorrowedAccountMeta, BorrowedInstruction, Instruction};
+use solana_instructions_sysvar::construct_instructions_data;
+use solana_pubkey::Pubkey;
+
+use crate::common::jupsol_fixtures_svc_suf;
+
+pub fn instructions_sysvar(instructions: &[Instruction], curr_idx: u16) -> (Pubkey, Account) {
+    let mut data = construct_instructions_data(
+        instructions
+            .iter()
+            .map(|instruction| BorrowedInstruction {
+                program_id: &instruction.program_id,
+                accounts: instruction
+                    .accounts
+                    .iter()
+                    .map(|meta| BorrowedAccountMeta {
+                        pubkey: &meta.pubkey,
+                        is_signer: meta.is_signer,
+                        is_writable: meta.is_writable,
+                    })
+                    .collect(),
+                data: &instruction.data,
+            })
+            .collect::<Vec<_>>()
+            .as_slice(),
+    );
+
+    *data.split_last_chunk_mut().unwrap().1 = curr_idx.to_le_bytes();
+
+    (
+        Pubkey::new_from_array(jiminy_sysvar_instructions::ID),
+        Account {
+            data,
+            owner: Pubkey::new_from_array(OWNER_ID),
+            lamports: 10_000_000,
+            executable: false,
+            rent_epoch: 0,
+        },
+    )
+}
+
+pub fn mock_empty_rebalance_record_account() -> Account {
+    const RR_RENT: u64 = Rent::DEFAULT.min_balance(std::mem::size_of::<RebalanceRecord>());
+    Account {
+        lamports: RR_RENT,
+        data: vec![0; std::mem::size_of::<RebalanceRecord>()],
+        owner: Pubkey::new_from_array(SYSTEM_PROGRAM_ID),
+        executable: false,
+        rent_epoch: 0,
+    }
+}
```

Comment:
this should be in `test-utils` crate

Reply:
actually, if you're testing the instruction chain properly, than `RebalanceRecord` should never exist beyond the instruction chain, so it should always just be `Default::default()`